playground/src/api/products.ts

```diff
@@ -26,7 +27,8 @@ export function getProductById(id: string | number, options?: Options<'json'>) {
   return products.get<ProductT>(id, options)
 }
 
-export async function changeProductAvailability(product: ProductListItem, options?: Options<'json'>) {
+export async function changeProductAvailability(product: ProductListItem, options?: Options<'json'>, _delay?: number) {
+  await delay(_delay ?? 0)
```

Comment:
let's not add the delay as an argument. We can always add it later to the inner function itself (randomized delay) or within the mutation fn